### PR TITLE
Started rendering the union of the shared blurs

### PR DESCRIPTION
**Pull request by gaaclarke**
issue: https://github.com/flutter/flutter/issues/157110

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: impeller/display_list/canvas.cc [hunk 1]
```diff
@@ -1115,24 +1116,37 @@ void Canvas::SaveLayer(const Paint& paint,
...
-            : Entity::RenderingMode::kSubpassAppendSnapshotTransform);
-
     if (will_cache_backdrop_texture) {
       FML_DCHECK(backdrop_data);
       // If all filters on the shared backdrop layer are equal, process the
       // layer once.
       if (backdrop_data->all_filters_equal &&
           !backdrop_data->shared_filter_snapshot.has_value()) {
-        // TODO(157110): compute minimum input hint.
+        Entity snapshot_entity;
```
Comment (jonahwilliams) [inline impeller/display_list/canvas.cc:1125]: I _think_ the coverage hint parameter of RenderToSnapshot accepts a source space rectangle. In which case, i think you can *just* uninion all the coverage rects and adjust the offset at the end.

**Comment by gaaclarke** [comment]
This doesn't quite work yet because the clip rect isn't considered.  Every time you query the filter's coverage it just gives you the full size of the input.  FilterContents has the method SetCoverageHint.  I'm not sure yet where the best place to capture the clip rect is yet and if we need to potentially store that in the BackdropData.

**Comment by jonahwilliams** [comment]
I think You need to use the input bounds to the filter saveLayer

**Comment by gaaclarke** [comment]
The TextFrameDispatcher [ignores the clip rect](https://github.com/gaaclarke/engine/blob/1fd6097102dc32e96f9aa6e1b6f99254454498ff/impeller/display_list/dl_dispatcher.h#L295).  In order to calculate the union we'd need to start tracking it there.

**Comment by jonahwilliams** [comment]
You don't need to track the clipRect to get a conservative union of the rects. Just use the saveLayer coverage

**Comment by gaaclarke** [comment]
> You don't need to track the clipRect to get a conservative union of the rects. Just use the saveLayer coverage

What are you talking about specifically?  I've looked everywhere in `TextFrameDispatcher::saveLayer` and there is no meaningful bounds anywhere.  All the values are just reporting the full viewport.

**Comment by gaaclarke** [comment]
Talked with Jonah chat and we've decided that we would need to track the clip rect.  It was decided that the extra complexity and computations may be better left off until we get some performance reports.

**Pull request closed without merging**